### ISSUE TITLE
`fundamental_from_projections` overflows to `inf` in float16 for pixel-unit projection matrices (5 of 9 entries)

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4877]
`fundamental_from_projections(P1, P2)` builds `F` from 4×4 determinants of rows of the two projection matrices and does not normalise the result. For pixel-unit matrices `P = K [R | t]` with focal lengths of 700–800 px the entries are of order 1e10 (`F[2,2] = 9.24e9` here), above float16's maximum of 65504: for float16 inputs the function computes the determinants in float32 (`fundamental.py:491-493`) and casts the result back (`fundamental.py:523`), so 5 of the 9 entries become `inf`. bfloat16 stays finite (relative error 1.2e-2) because it has float32's exponent range, and the same geometry with normalised-camera matrices `[R | t]` stays finite in float16. `F` is defined only up to scale, so any scale that fits the dtype describes the same epipolar geometry; a user expects a finite `F`.

## Repro

```python
import torch
from kornia.geometry.conversions import axis_angle_to_rotation_matrix
from kornia.geometry.epipolar import (essential_from_Rt, find_fundamental, fundamental_from_essential,
                                      fundamental_from_projections)

K1 = torch.tensor([[800.0, 0, 320], [0, 760, 200], [0, 0, 1]], dtype=torch.float64)
K2 = torch.tensor([[700.0, 0, 300], [0, 740, 240], [0, 0, 1]], dtype=torch.float64)
R = axis_angle_to_rotation_matrix(torch.tensor([[0.1, -0.2, 0.05]], dtype=torch.float64))[0]
t = torch.tensor([[0.5], [0.1], [0.02]], dtype=torch.float64)
I, z = torch.eye(3, dtype=torch.float64), torch.zeros(3, 1, dtype=torch.float64)
P1 = (K1 @ torch.cat([I, z], 1))[None]   # pixel-unit projection matrices
P2 = (K2 @ torch.cat([R, t], 1))[None]

F64 = fundamental_from_projections(P1, P2)
print("float64: F[2,2] =", f"{F64[0, 2, 2].item():.4g}", " ||F|| =", f"{F64.norm().item():.4g}")
for dt in (torch.float32, torch.bfloat16, torch.float16):
    F = fundamental_from_projections(P1.to(dt), P2.to(dt))
    fin = torch.isfinite(F)
    rel = ((F.double() - F64).norm() / F64.norm()).item() if fin.all() else float("nan")
    print(f"{str(dt):15s} inf entries: {int((~fin).sum())} of 9, rel err vs float64: {rel:.2g}")
# the same geometry with unit-scale (normalised-camera) projection matrices stays finite in float16
Pn1, Pn2 = torch.cat([I, z], 1)[None], torch.cat([R, t], 1)[None]
print("normalised P, float16, all finite:", torch.isfinite(fundamental_from_projections(Pn1.half(), Pn2.half())).all().item())

# the three F producers return the same epipolar geometry on three different scales
torch.manual_seed(0)
X = torch.rand(12, 3, dtype=torch.float64) * 2 - 1
X[:, 2] += 5
x1 = (K1 @ X.T).T
x2 = (K2 @ (R @ X.T + t)).T
x1, x2 = (x1[:, :2] / x1[:, 2:])[None], (x2[:, :2] / x2[:, 2:])[None]
Fe = fundamental_from_essential(essential_from_Rt(I[None], z[None], R[None], t[None]), K1[None], K2[None])
Ff = find_fundamental(x1, x2)
for name, F in [("fundamental_from_projections", F64), ("fundamental_from_essential", Fe), ("find_fundamental", Ff)]:
    r = F / Ff
    print(f"{name:29s} F[2,2] = {F[0, 2, 2].item():+.4g}   F / find_fundamental, all 9 entries: {r.min().item():+.6g} .. {r.max().item():+.6g}")
```

Output:

```text
float64: F[2,2] = 9.238e+09  ||F|| = 9.243e+09
torch.float32   inf entries: 0 of 9, rel err vs float64: 8.7e-08
torch.bfloat16  inf entries: 0 of 9, rel err vs float64: 0.012
torch.float16   inf entries: 5 of 9, rel err vs float64: nan
normalised P, float16, all finite: True
fundamental_from_projections  F[2,2] = +9.238e+09   F / find_fundamental, all 9 entries: +9.23813e+09 .. +9.23813e+09
fundamental_from_essential    F[2,2] = -0.02933   F / find_fundamental, all 9 entries: -0.0293326 .. -0.0293326
find_fundamental              F[2,2] = +1   F / find_fundamental, all 9 entries: +1 .. +1
```

kornia `main` 7ddf731bd, torch 2.14.0 (and 2.5.1: same), CPU float64 input cast to float32 / bfloat16 / float16, macOS arm64. The last three lines show the same epipolar geometry on three scales: `fundamental_from_projections` returns 9.24e9 × `find_fundamental`'s `F`, and `fundamental_from_essential(essential_from_Rt(...))` returns −0.0293 × it (opposite sign).

## Classification

Clearly broken output. No idiom of 1–9; candidate: an unnormalised intermediate overflowing a narrow dtype (`inf` returned silently, as in idiom 4).

## Disposition

MERGE IN WINDOW — behaviour change: a fix that normalises (the rows of `P1`, `P2` before the determinants, or `F` itself before the cast back) changes the output scale contract for every dtype, not only float16: code that uses today's scale (proportional to products of `P` entries, 9.24e9 here) gets a different number. `fundamental_from_essential` (unnormalised, the scale of `E` and `K`) and `find_fundamental` (`F[2,2] = 1`) already return `F` on different scales today, so the fix should choose the contract for the producers together rather than for this function alone.

Found by the batch-8 conventions audit (two-view geometry); a `test_wart_*` pin in the docs PR flips when the fix lands.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
